File: medplum-java/medplum-java-model/src/main/java/com/fhir/ElementDefinition_Discriminator.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableElementDefinition_Discriminator.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface ElementDefinition_Discriminator {
  Optional<String> id();

  Optional<Elementdefinition_discriminatorType> type();

  Optional<String> path();

  Optional<List<Extension>> modifierExtension();

  Optional<List<Extension>> extension();

  static ImmutableElementDefinition_Discriminator.Builder builder() {
    return ImmutableElementDefinition_Discriminator.builder();
  }
}

File: medplum-typescript/src/model/fhir/ElementDefinition_Discriminator.ts
import { Elementdefinition_discriminatorType } from "./Elementdefinition_discriminatorType";
import { Extension } from "./Extension";

export interface ElementDefinition_Discriminator {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    type: Elementdefinition_discriminatorType | null | undefined;
    path: string | null | undefined;
}
